packages/args.ol

```diff
@@ -0,0 +1,115 @@
+from string-utils import StringUtils
+from console import Console
+
+
+type GetParsedArgsRequest {
+    args*: string 
+    argsMap*: void {
+        name: string        // name of the expected argument
+        optional: bool      // specifies if the argument is optional
+        description: string // describes the argument
+        isFlag: bool       // if true, the argument is a flag (e.g. --verbose)
+    }
+
+}
+
+type GetParsedResponse {
+    parsedArgs* {
+        name: string 
+        value: string | void
+    }
+}
+
+interface ArgsInterface {
+    RequestResponse:
+        getParsedArgs( GetParsedArgsRequest )( GetParsedResponse ) throws ParametersError( string ) Help
```

Comment:
`ParametersError` is a bit surprising since parameters hasn't been mentioned before, maybe `ParsingError`?